.github/actions/setup-elixir/action.yml

```diff
@@ -0,0 +1,49 @@
+name: "Setup Elixir"
+description: "Sets up the correct Elixir version and installs deps"
+inputs:
+  mix_env:
+    description: "Optionally limit deps to mix env"
+    type: string
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Tool Versions
+      id: versions
+      uses: marocchino/tool-versions-action@v1.2.0
+    - uses: erlef/setup-beam@v1
+      id: setup-beam
+      with:
+        otp-version: ${{ steps.versions.outputs.erlang }}
+        elixir-version: ${{ steps.versions.outputs.elixir }}
+    - uses: actions/cache/restore@v4
+      name: Restore Elixir Deps Cache
+      id: cache
+      with:
+        path: |
+          elixir/deps
+          elixir/_build/${{ inputs.mix_env }}
+        key: ${{ runner.os }}-${{ steps.setup-beam.outputs.elixir-version }}-${{ hashFiles('elixir/mix.lock') }}
+    - name: Install Dependencies
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      shell: bash
+      working-directory: ./elixir
+      run: |
+        if [[ -z "${{ inputs.mix_env }}" ]]; then
+          mix deps.get
+        else
+          mix deps.get --only ${{ inputs.mix_env }}
+        fi
+    - name: Compile Dependencies
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      shell: bash
+      working-directory: ./elixir
+      run: mix deps.compile --skip-umbrella-children
```

Comment:
Since those dependencies are compiled the cache key must have `MIX_ENV` as part of it.

Reply:
Good catch, fixed